static/js/muting.js

```diff
@@ -4,14 +4,15 @@ var exports = {};
 
 var muted_topics = new Dict({fold_case: true});
 
-exports.mute_topic = function (stream, topic) {
+exports.mute_topic = function (stream, topic, manual_trigger) {
     var sub_dict = muted_topics.get(stream);
     if (!sub_dict) {
         sub_dict = new Dict({fold_case: true});
         muted_topics.set(stream, sub_dict);
     }
     sub_dict.set(topic, true);
     unread.update_unread_counts();
+    muting_ui.mute_notification(stream, topic, manual_trigger);
```

Comment:
Yep. I'll move it. Also currently reworking stuff because front-end tests
failed. Will push with updates soon!

On Thu, Aug 25, 2016 at 5:07 PM, showell <notifications@github.com> wrote:

> In static/js/muting.js
> <https://github.com/zulip/zulip/pull/1711#discussion_r76346937>:
>
> >      var sub_dict = muted_topics.get(stream);
> >      if (!sub_dict) {
> >          sub_dict = new Dict({fold_case: true});
> >          muted_topics.set(stream, sub_dict);
> >      }
> >      sub_dict.set(topic, true);
> >      unread.update_unread_counts();
> > +    muting_ui.mute_notification(stream, topic, manual_trigger);
>
> Can we let either ui.js or muting_ui.js orchestrate the notification? I am
> hoping to keep muting.js mostly free of UI dependencies.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zulip/zulip/pull/1711/files/73db5870f36bb6baaf88ee8a9f4455216742adf1..13b6592148d039c13f4046aa0c4f507935207fd9#r76346937>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJ199wglsTSOw9HpnDWZ5BP64k6K-Fxyks5qji5GgaJpZM4Jtnrn>
> .
>